packages/babel-plugin/src/class-names/__tests__/behaviour.test.tsx

```diff
@@ -22,86 +22,226 @@ describe('class names behaviour', () => {
+        <ClassNames>
+          {({ css }) => children(css({ fontSize: '20px' }))}
+        </ClassNames>
+      );
+   `);
+
+    expect(actual).toMatchInlineSnapshot(`
+      "/* File generated by @compiled/babel-plugin v0.0.0 */
+      import * as React from 'react';
+      import { ax, CC, CS } from \\"@compiled/react/runtime\\";
+      const _ = \\"._1wybgktf{font-size:20px}\\";
+
+      const ListItem = ({
+        children
+      }) => <CC>
+          <CS>{[_]}</CS>
+          {children(\\"_1wybgktf\\")}
+        </CC>;"
+    `);
   });
 
-  xit('should set children as function into a jsx expression', () => {
+  it('should place self closing jsx element as a child', () => {
     const actual = transform(`
     import { ClassNames } from '@compiled/react';
 
     const ZoomOnHover = ({ children }) => (
       <ClassNames>
-        {({ css }) =>
-          children({
-            className: css({
-              transition: 'transform 2000ms',
-              ':hover': {
-                transform: 'scale(2)',
-              },
-            }),
-          })
-        }
+        {({ css }) => <div className={css({ fontSize: 12 })} />}
       </ClassNames>
     );
   `);
 
-    expect(actual).toInclude(`{children({
-    className: "css-test",
-})}`);
+    expect(actual).toInclude(`<div className={\"_1wyb1fwx\"} /`);
   });
 
-  xit('should place self closing jsx element as a child', () => {
+  it('should replace style identifier with undefined', () => {
     const actual = transform(`
-    import { ClassNames } from '@compiled/react';
+      import { ClassNames } from '@compiled/react';
 
-    const ZoomOnHover = ({ children }) => (
-      <ClassNames>
-        {({ css }) => <div className={css({ fontSize: 12 })} />}
-      </ClassNames>
-    );
+      const Component = ({ children }) => (
+        <ClassNames>
+          {({ css, style }) => <div style={style} className={css({ fontSize: 12 })} />}
+        </ClassNames>
+      );
   `);
 
-    expect(actual).toInclude(`</CS><div className={\"css-test\"}/></CC>`);
+    expect(actual).toInclude(`style={undefined}`);
+  });
+
+  it('should replace style identifier with css variable object', () => {
+    const actual = transform(`
+      import { ClassNames } from '@compiled/react';
+
+      const Component = ({ children, color }) => (
+        <ClassNames>
+          {({ css, style }) => <div style={style} className={css({ color })} />}
+        </ClassNames>
+      );
+  `);
+
+    expect(actual).toMatchInlineSnapshot(`
+      "/* File generated by @compiled/babel-plugin v0.0.0 */
+      import * as React from 'react';
+      import { ax, CC, CS } from \\"@compiled/react/runtime\\";
+      const _ = \\"._syaz1aj3{color:var(--_1ylxx6h)}\\";
+
+      const Component = ({
+        children,
+        color
+      }) => <CC>
+          <CS>{[_]}</CS>
+          {<div style={{
+          \\"--_1ylxx6h\\": color
+        }} className={\\"_syaz1aj3\\"} />}
+        </CC>;"
+    `);
```

Comment:
Uncommon case. See if we want to handle it. I checked. Its not working;

```ts
<ClassNames>
{(arg) => {
  const { css } = arg;

  return (
    <div
      className={css`
        font-size: ${props.size}px;
        color: red;
      `}>
      hello world
    </div>
  );
}}
</ClassNames>
```

Reply:
nice, yeah i'll add this to the follow up ticket!

Reply:
found a edge case that isnt statically extracted

```js
      import { ClassNames } from '@compiled/react';

      const ListItem = () => (
        <ClassNames>
          {({ css }) => {
            const color = 'blue';
            return <div className={css({ fontSize: '20px', color })}>hello, world!</div>;
            // color ends up being a css variable
          }}
        </ClassNames>
      );
```

Reply:
made issue